### DISCUSSION TITLE
Is it possible to change the protocol wails://?

### DISCUSSION THREAD

**Discussion by aledbf** [wailsapp/wails#1747] [Q&A] [answered]
to be like windows `http://wails.localhost`?

I am trying to add auth0 and only supports http/s as a scheme.
Thanks!

**Comment by stffabi** [comment]
Unfortunately that's not possible on all platforms. On Windows we use `http://wails.localhost` because we can't yet register a custom scheme there. On MacOS e.g. there's no other way to serve content from the application to the WKWebView without registering a custom scheme. AFAIK the same applies to WebKit2GTK. 
Using http on those platforms would mean we need to start a localhost http server to serve assets, which would have some other drawbacks.

I've just briefly taken a look into the auth0 documentations and it seems like they support using custom schemes for native apps, according to the screenshot [Application-URIs](https://auth0.com/docs/get-started/applications/application-settings#application-uris).
> Make sure to specify the protocol (https: // ) otherwise the callback may fail in some cases. With the exception of custom URI schemes for native clients, all callbacks should use protocol https: //.

**Comment by aledbf** [comment, marked as answer]
@stffabi thank you for the quick response.

I found an approach in [Slack that works](https://gophers.slack.com/archives/CJ4P9F7MZ/p1650366744590869)

Basically, start a local server to process the callback and reload the application

``` 
http.HandleFunc("/callback", func(w http.ResponseWriter, r *http.Request) {
		// check errors

		runtime.WindowReloadApp(ctx)
})
```
